src/System.IO.FileSystem/tests/Directory/Move_str_str.cs

```diff
@@ -1,430 +1,281 @@
+            string testDir = GetTestFilePath();
+            string testDirSource = Path.Combine(testDir, "*");
+            string testDirShouldntMove = Path.Combine(testDir, "*t");
+            string testDirDest = Path.Combine(testDir, "*" + GetTestFileName());
+
+            Directory.CreateDirectory(testDirSource);
+            Directory.CreateDirectory(testDirShouldntMove);
+            Move(testDirSource, testDirDest);
+            Assert.True(Directory.Exists(testDirDest));
+            Assert.False(Directory.Exists(testDirSource));
+            Assert.True(Directory.Exists(testDirShouldntMove));
+
+            Move(testDirDest, testDirSource);
+            Assert.False(Directory.Exists(testDirDest));
+            Assert.True(Directory.Exists(testDirSource));
+            Assert.True(Directory.Exists(testDirShouldntMove));
+        }
 
-            // [] PathTooLongException if destination name is too long
-            //-----------------------------------------------------------------
-            strLoc = "Loc_00044";
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsWhitespacePath()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            Assert.Throws<ArgumentException>(() => Move(testDir.FullName, "         "));
+            Assert.Throws<ArgumentException>(() => Move(testDir.FullName, "\n"));
+            Assert.Throws<ArgumentException>(() => Move(testDir.FullName, ""));
+            Assert.Throws<ArgumentException>(() => Move(testDir.FullName, ">"));
+            Assert.Throws<ArgumentException>(() => Move(testDir.FullName, "<"));
+            Assert.Throws<ArgumentException>(() => Move(testDir.FullName, "\0"));
+            Assert.Throws<ArgumentException>(() => Move(testDir.FullName, "\t"));
+        }
 
-            String str = new string('a', IOInputs.MaxPath);
-            iCountTestcases++;
-            try
-            {
-                Directory.Move(TestInfo.CurrentDirectory, str);
-                iCountErrors++;
-                printerr("Error_00045! Expected exception not thrown");
-            }
-            catch (PathTooLongException)
-            {
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_00047! Incorrect exception thrown, exc==" + exc.ToString());
-            }
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        [ActiveIssue(2205)]
+        public void UnixWhitespacePath()
+        {
+            string testDir = GetTestFilePath() + Path.DirectorySeparatorChar;
+            string testDirToMove = Path.Combine(testDir, GetTestFileName());
+            Directory.CreateDirectory(testDirToMove);
+            Move(testDirToMove, testDir + "         ");
+            Move(testDir + "         ", testDir + "\n");
+            Move(testDir + "\n", testDir + "\t");
+            Move(testDir + "\t", testDir + "");
```

Comment:
Seems like this should fail because we're trying to move the subdirectory to overwrite the parent (the string being appended is the empty string).